lib/util/DateTimeHelpers.ts

```diff
@@ -0,0 +1,191 @@
+import type {
+  IDateRepresentation, IDateTimeRepresentation, IDayTimeDurationRepresentation,
+  IDurationRepresentation,
+  ITimeRepresentation,
+  ITimeZoneRepresentation, IYearMonthDurationRepresentation,
+} from './InternalRepresentations';
+import { extractDayTimeDur } from './InternalRepresentations';
+import { maximumDayInMonthFor } from './specAlgos';
+
+function numSer(num: number, min = 2): string {
+  return num.toLocaleString(undefined, { minimumIntegerDigits: min, useGrouping: false });
+}
+
+// Parsers:
+export function dateTimeParser(dateTimeStr: string, errorCreator?: () => Error): IDateTimeRepresentation {
```

Comment:
`dateTimeParser` (and similar below) are a bit confusing, as it sounds like this is an object, while it is just a function.
Something like `dateTimeParse` or `parseDateTime` might be better names.